src/main/resources/META-INF/resources/primefaces-extensions/fuzzysort/primefaces.fuzzysort.js

```diff
@@ -0,0 +1,42 @@
+PrimeFaces.widget.Fuzzysort = PrimeFaces.widget.BaseWidget.extend({
+
+    init: function (cfg) {
+        this.cfg = cfg;
+        this.id = this.cfg.id;
+        this.jqId = PrimeFaces.escapeClientId(this.id);
+//        this.jq = $(PrimeFaces.escapeClientId(this.cfg.target));
+
+        this.input = $(PrimeFaces.escapeClientId(this.cfg.id + "_fuzzysort-search-box"));
+        this.results = $(PrimeFaces.escapeClientId(this.cfg.id + "_fuzzysort-search-results"));
+        this.datasource = JSON.parse(this.cfg.value);
+        
+        //Visual effects
+        PrimeFaces.skinInput(this.input);
+
+        var $this = this;
+//        console.log($this);
+
+        jQuery(document).ready(function () {
+            $this.search();
+        });
+
+        // Run a search on input change
+        $this.input.on('input', $this.search);
+    },
+
+    search: function () {
+        this.results.empty();
```

Comment:
First line of search should be `var $this = this;`

Reply:
When I put `var $this = this;`, `$this` gives input component only. I cannot access results and datasource variables.

Reply:
I have removed `document.ready` since the list has already populated in `FuzzysortRenderer`. 
I couldn't make it work search function. So the codes are working in `init` as jquery event handler.

Reply:
created search function and finally it is working